pkg/scheduler/eventhandlers.go

```diff
@@ -492,37 +493,54 @@ func nodeSpecUnschedulableChanged(newNode *v1.Node, oldNode *v1.Node) bool {
 }
 
 func preCheckForNode(nodeInfo *framework.NodeInfo) queue.PreEnqueueCheck {
-	// In addition to the checks in kubelet (pkg/kubelet/lifecycle/predicate.go#GeneralPredicates),
-	// the following logic appends a taint/toleration check.
-	// TODO: verify if kubelet should also apply the taint/toleration check, and then unify the
-	// logic with kubelet and move to a shared place.
-	//
 	// Note: the following checks doesn't take preemption into considerations, in very rare
 	// cases (e.g., node resizing), "pod" may still fail a check but preemption helps. We deliberately
 	// chose to ignore those cases as unschedulable pods will be re-queued eventually.
 	return func(pod *v1.Pod) bool {
-		if len(noderesources.Fits(pod, nodeInfo, feature.DefaultFeatureGate.Enabled(features.PodOverhead))) != 0 {
-			return false
-		}
+		nonResources, insufficientResources := GeneralFilter(pod, nodeInfo, false)
+		return len(nonResources) == 0 && len(insufficientResources) == 0
+	}
+}
 
-		// Ignore parsing errors for backwards compatibility.
-		matches, _ := nodeaffinity.GetRequiredNodeAffinity(pod).Match(nodeInfo.Node())
-		if !matches {
-			return false
+// GeneralFilter checks whether the pod meets noderesources/nodeport/nodeAffinity/tainttoleration/nodename and returns the reason
+// Used in kubelet(pkg/kubelet/lifecycle/predicate.go) and scheduler
+// includeAllFailures indicates whether we need to get all the reasons.
+// In the scheduler, we only need to fail quickly, so all the reasons are not needed, and kubelet is needed.
```

Comment:
```suggestion
// GeneralFilter calls the filtering logic of 
// noderesources/nodeport/nodeAffinity/tainttoleration/nodename
// and returns the failure reasons.
// It's used in kubelet(pkg/kubelet/lifecycle/predicate.go) and scheduler.
// It returns the first failure if `includeAllFailures` is set to false; otherwise
// returns all failures.
```